### PR TITLE
Added cardinal directions buttons

### PR DESCRIPTION
**Pull request by JordanNg**
I added the cardinal direction buttons to the UI and set it up so that each time a direction is pressed the ships will move. I still need to implement the movement of the ships. I also made it so that the ships can only move twice per game.

### CODE REVIEW

File: src/main/java/assets/css/custom.css [hunk 1]
```diff
@@ -286,3 +286,83 @@ position:fixed;
...
+   border: none;
+   border-radius: 15px;
+   box-shadow: 0 9px #999;
+     position:fixed;
+     left:100px;
+ position:fixed;
+ left:160px;
+ top:340px;
+ }
+
```
Comment (ramosjos) [inline src/main/java/assets/css/custom.css]: CSS looks good. Does the top:340px cram the left side of the game or is their still enough space for all the buttons. Will test.

File: src/main/java/assets/css/custom.css [hunk 2]
```diff
@@ -286,3 +286,83 @@ position:fixed;
...
+   box-shadow: 0 9px #999;
+     position:fixed;
+     left:100px;
+ position:fixed;
+ left:160px;
+ top:400px;
+ }
+
+ .east{
+   display: none;
```
Comment (ramosjos) [inline src/main/java/assets/css/custom.css]: Never mind, display style is none so they only appear after all ships have been placed and user sinks two ships. Nice call to not clutter the screen.

**Comment by phillist** [comment]
Looks clean, however we need to have test and to check ships location for edge of map. because if a ship is near the edge it shouldn't be moved, however the rest of the fleet should still be able to move by themselves. and remember, we are adding a submarine, make sure this is able to operate with the submarine as well.

File: src/main/java/assets/css/custom.css [hunk 3]
```diff
@@ -286,3 +286,83 @@ position:fixed;
...
+   background-color: #0e28a5;
+   border: none;
+   border-radius: 15px;
+   box-shadow: 0 9px #999;
+     position:fixed;
+     left:100px;
+ position:fixed;
+ left:100px;
+ top:400px;
+ }
```
Comment (ramosjos) [inline src/main/java/assets/css/custom.css]: All the css looks fine and is only visible when the ship buttons disappear so the screen is not cluttered.

File: src/main/java/assets/game.js [hunk 4]
```diff
@@ -27,9 +27,56 @@ function showMoveFleetButton(){
...
+    document.getElementById("move_fleet").style.display = "none";
+}
+
+function hideCardinalDirection() {
+    document.getElementById("N").style.display = "none";
+    document.getElementById("S").style.display = "none";
+    document.getElementById("E").style.display = "none";
+    document.getElementById("W").style.display = "none";
+}
+
```
Comment (ramosjos) [inline src/main/java/assets/game.js]: Great idea to hide the cardinal directions once they are no longer needed.

File: src/main/java/views/ApplicationController/index.ftl.html [hunk 5]
```diff
@@ -19,6 +19,10 @@ <h1 align="center" >Battleship</h1>
     <ul><button align="left" id="place_submarine" class="submov">Place Submarine</button></ul>
     <ul><button id="place_sonar" class="sonmov" class="button">Place Sonar Pulse</button></ul>
     <ul><button id="move_fleet" class="fleetmov" class="button">Move Fleet</button></ul>
+    <ul><button id="N" class="north" class="button">N</button></ul>
+    <ul><button id="S" class="south" class="button">S</button></ul>
+    <ul><button id="E" class="east" class="button">E</button></ul>
+    <ul><button id="W" class="west" class="button">W</button></ul>
     <ul class="vermov"><input align="left" type="checkbox" id="is_vertical"  style="zoom:200%;color:blue"><p class="vertical">Vertical</p></ul>
```
Comment (ramosjos) [inline src/main/java/views/ApplicationController/index.ftl.html]: I like the simplicity. of N S W E

File: src/main/java/assets/game.js [hunk 6]
```diff
@@ -48,10 +95,23 @@ function markHits(board, elementId, surrenderText) {
...
     });
-    if(shipIsSunk === "true")
+
+    if(shipIsSunk === "true") {
         showSolarPulseButton();
-    if(moveFleet >= 2)
+    }
+    
+    if(moveFleet >= 2) {
         showMoveFleetButton();
```
Comment (ramosjos) [inline src/main/java/assets/game.js]: Nice implementation and creativity for limiting the user to two movements.

File: src/main/java/cs361/battleships/models/Square.java [hunk 7]
```diff
@@ -51,6 +51,13 @@ public int getRow() {
 		return row;
 	}
 
+	public void setColumn(char newCol) {
+		column = newCol;
+	}
+
+	public void setRow(int newRow) {
+		row = newRow;
+	}
```
Comment (ramosjos) [inline src/main/java/cs361/battleships/models/Square.java]: Great job implementing a simple fix for implementing the move fleet.

File: src/main/java/assets/game.js [hunk 8]
```diff
@@ -48,10 +95,23 @@ function markHits(board, elementId, surrenderText) {
...
+        document.getElementById("move_fleet").addEventListener("click", function() {
+            if (fuel > 0) {
+                showCardinalDirection();
+                fuel = fuel - 1;
+            } else {
+                hideMoveFleetButton();
+                alert("ALERT: Your ships do not have enough fuel to move.");
```
Comment (ramosjos) [inline src/main/java/assets/game.js]: When will the hideMoveFleetButton() actually be called. Wouldn't this leave it on the UI until the User pressed the move fleet button a 3rd time? a better idea would be 

    document.getElementById("move_fleet").addEventListener("click", function() {
            if (fuel > 0) {
                showCardinalDirection();
                fuel = fuel - 1;
                    if(fuel == 0){
                    hideMoveFleetButton();
            } else {
                alert("ALERT: Your ships do not have enough fuel to move.");


This way the button is hidden right after they press the move fleet button a second time?

Comment (ramosjos) [reply src/main/java/assets/game.js]: Syntax is not right but you get the point

**Comment by ramosjos** [comment]
Tests need to be included with the pr's. But since the move fleet is not finished, not really anything to test yet. Tests and Travis passed, will merge.

**Comment by ramosjos** [comment]
Accidentally closed it. Idk why have now reopened.
